### PR TITLE
improve rules that often match

### PR DESCRIPTION
**Pull request by mr-tz**
I've analyzed the top 50 or so rule matches over 5000 recent random samples.
In these rules I noticed room for improvement. Happy to discuss as needed.

### CODE REVIEW

File: executable/resource/extract-resource-via-kernel32-functions.yml [hunk 1]
```diff
@@ -28,4 +28,3 @@ rule:
             - api: LdrFindResourceEx_U
           - api: kernel32.SizeofResource
           - api: kernel32.FreeResource
-      - api: user32.LoadString
```
Comment (mr-tz) [inline executable/resource/extract-resource-via-kernel32-functions.yml:31]: I don't think that's really what we are after normally. Anyone thing it's worth creating a new rule for it?

File: host-interaction/process/terminate/terminate-process-via-fastfail.yml [hunk 2]
```diff
@@ -1,17 +0,0 @@
-rule:
```
Comment (mr-tz) [inline host-interaction/process/terminate/terminate-process-via-fastfail.yml:1]: ref: #143, I've only seen this hit in library code (which Python flirt often doesn't recognize)

File: host-interaction/registry/create-or-open-registry-key.yml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 rule:
   meta:
     name: create or open registry key
```
Comment (mr-tz) [inline host-interaction/registry/create-or-open-registry-key.yml:3]: analogous to `create or open file`, we have a lot more specific rules that use this (now library) rule, e.g. `set registry value` or `delete registry value`

File: executable/resource/extract-resource-via-kernel32-functions.yml (hunk 1 above)
Comment (williballenthin) [reply executable/resource/extract-resource-via-kernel32-functions.yml:31]: agree that this should be removed

File: lib/calculate-modulo-256-via-x86-assembly.yml [hunk 4]
```diff
@@ -15,5 +15,5 @@ rule:
     - and:
       - mnemonic: and
       - or:
-        - number: 0x800000FF
-        - number: 0xFF
+        - operand[1].number: 0x800000FF
+        - operand[1].number: 0xFF
```
Comment (williballenthin) [inline lib/calculate-modulo-256-via-x86-assembly.yml:19]: can you explain this change?

**Review by williballenthin — APPROVED** [review]

File: lib/calculate-modulo-256-via-x86-assembly.yml (hunk 4 above)
Comment (mr-tz) [reply lib/calculate-modulo-256-via-x86-assembly.yml:19]: trying to be more explicit here, but yeah, there can only be an immediate in the second operand, so I guess this is not useful here

Comment (mr-tz) [reply lib/calculate-modulo-256-via-x86-assembly.yml:19]: any preference? be explicit here or undo change and stick to `number`?

Comment (mr-tz) [inline lib/calculate-modulo-256-via-x86-assembly.yml:19]:
```suggestion
        - operand: 0x800000FF
        - operand: 0xFF
```

File: lib/calculate-modulo-256-via-x86-assembly.yml [hunk 5]
```diff
@@ -15,5 +15,5 @@ rule:
     - and:
       - mnemonic: and
       - or:
-        - number: 0x800000FF
-        - number: 0xFF
+        - operand: 0x800000FF
+        - operand: 0xFF
```
Comment (mr-tz) [inline lib/calculate-modulo-256-via-x86-assembly.yml:19]:
```suggestion
        - number: 0x800000FF
        - number: 0xFF
```

File: host-interaction/file-system/exists/check-if-file-exists.yml [hunk 6]
```diff
@@ -17,12 +17,12 @@ rule:
           - api: kernel32.GetFileAttributes
           - instruction:
             - mnemonic: cmp
-            - number: 0xFFFFFFFF = INVALID_FILE_ATTRIBUTES
+            - operand[1].number: 0xFFFFFFFF = INVALID_FILE_ATTRIBUTES
```
Comment (mr-tz) [inline host-interaction/file-system/exists/check-if-file-exists.yml:20]:
```suggestion
            - number: 0xFFFFFFFF = INVALID_FILE_ATTRIBUTES
```

**Review by williballenthin — APPROVED** [review]

**Pull request merged**